torchaudio/datasets/folder.py

```diff
@@ -0,0 +1,73 @@
+import os
+from typing import Tuple
+import torchaudio
+from torch.utils.data import Dataset
+from torch import Tensor
+from pathlib import Path
+
+def load_audio_item(filepath: str, path: str, label_name: str) -> Tuple[Tensor, int, str, str]:
+    relpath = os.path.relpath(filepath, path)
+    label, filename = os.path.split(relpath)
+    if label_name is not None:
+        label = label_name
+    waveform, sample_rate = torchaudio.load(filepath)
+    return waveform, sample_rate, label, filename
+
+class AudioFolder(Dataset):
+    """Create a Dataset from Local Files.
+
+    Args:
+        path (str): Path to the directory where the dataset is found or downloaded.
+        suffix (str) : Audio file type, defaulted to ".WAV".
+        pattern (str) : Find pathnames matching this pattern. Defaulted to "*/*" 
+        new_sample_rate (int) : Resample audio to new sample rate specified.
+        spectrogram_transform (bool): If `True` transform the audio waveform and returns it  
+        transformed into a spectrogram tensor.
+        label (str): The label is pulled from the folders in the path, this allows you to statically
+        define the label string.
+    """
+
+
+    def __init__(
+            self,
+            path: str,
+            suffix: str = ".wav",
+            pattern: str = "*/*",
+            new_sample_rate: int = None,
+            spectrogram_transform: bool = False,
+            label: str = None
```

Comment:
If this label is provided by user, the Dataset always returns the same label. In that case, users can always use their own label, so there is no benefit making this into a parameter.

Reply:
If left null then it doesn't set the label or pulls it from the subfolders. If I am pulling all one class from a folder then label comes back as an empty string. It is useful to be able to define it when creating a dataset of one class from one folder.